shared/src/conversions.rs

```diff
@@ -1,12 +1,20 @@
-use num::bigint::Sign;
+use anyhow::{anyhow, Result};
 use num::BigInt;
+use num::{bigint::Sign, Zero};
 use num::{BigRational, ToPrimitive as _};
 use primitive_types::U256;
 
 pub fn big_rational_to_float(ratio: &BigRational) -> Option<f64> {
     Some(ratio.numer().to_f64()? / ratio.denom().to_f64()?)
 }
 
+pub fn big_rational_to_u256(ratio: &BigRational) -> Result<U256> {
+    if ratio.denom() == &BigInt::zero() {
+        return Err(anyhow!(" Division by 0 in BigRational to U256 conversion"));
```

Comment:
```suggestion
        return Err(anyhow!("Division by 0 in BigRational to U256 conversion"));
```

or even better, you can use the `ensure!` macro, which is an `assert` which errors instead of panicking:
```rust
    ensure!(ratio.denom() == &BigInt::zero(), "Division by 0 in BigRational to U256 conversion");
```